build-android

```diff
@@ -0,0 +1,139 @@
+	
+        pushd $SWIFT_PATH
+            pushd swift-corelibs-foundation
+                # Libfoundation script is not completely prepared to handle cross compilation yet.
+                ln -sf $SWIFT_ANDROID_BUILD_PATH/swift-linux-x86_64/lib/swift $ANDROID_STANDALONE_SYSROOT/usr/lib/
+                cp $SWIFT_ANDROID_BUILD_PATH/swift-linux-x86_64/lib/swift/android/armv7/* $SWIFT_ANDROID_BUILD_PATH/swift-linux-x86_64/lib/swift/android/
+
+                # Search path for curl seems to be wrong in foundation
+                #cp -r .build/openssl-1.0.2l/ssl $ANDROID_STANDALONE_SYSROOT/usr/include
+                cp -r .build/curl/include/curl $ANDROID_STANDALONE_SYSROOT/usr/include
+                ln -s $ANDROID_STANDALONE_SYSROOT/usr/include/curl $ANDROID_STANDALONE_SYSROOT/usr/include/curl/curl
+                ln -s $ANDROID_STANDALONE_SYSROOT/usr/include/libxml2/libxml $ANDROID_STANDALONE_SYSROOT/usr/include/libxml
+ 
+                env \
+                    SWIFTC="$SWIFT_ANDROID_BUILD_PATH/swift-linux-x86_64/bin/swiftc" \
+                    CLANG="$SWIFT_ANDROID_BUILD_PATH/llvm-linux-x86_64/bin/clang" \
+                    SWIFT="$SWIFT_ANDROID_BUILD_PATH/swift-linux-x86_64/bin/swift" \
+                    SDKROOT="$SWIFT_ANDROID_BUILD_PATH/swift-linux-x86_64" \
+                    BUILD_DIR="$SWIFT_ANDROID_BUILD_PATH/foundation-linux-x86_64" \
+                    DSTROOT="/" \
+                    PREFIX="/usr" \
+                    CFLAGS="-DDEPLOYMENT_TARGET_ANDROID -DDEPLOYMENT_ENABLE_LIBDISPATCH --sysroot=$ANDROID_NDK_PATH/platforms/android-22/arch-arm -I$ANDROID_ICU_PATH/armeabi-v7a/include -I${SDKROOT}/lib/swift -I$ANDROID_NDK_PATH/sources/android/support/include -I$ANDROID_STANDALONE_SYSROOT/usr/include -I$SWIFT_PATH/swift-corelibs-foundation/closure" \
+                    SWIFTCFLAGS="-DDEPLOYMENT_TARGET_ANDROID -DDEPLOYMENT_ENABLE_LIBDISPATCH -I$ANDROID_NDK_PATH/platforms/android-22/arch-arm/usr/include" \
+                    LDFLAGS="-fuse-ld=gold --sysroot=$ANDROID_NDK_PATH/platforms/android-22/arch-arm -L$ANDROID_NDK_PATH/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x -L$ANDROID_ICU_PATH/armeabi-v7a -L$ANDROID_STANDALONE_SYSROOT/usr/lib -ldispatch " \
+                    ./configure \
+                        Release \
+                        --target=armv7-none-linux-androideabi \
+                        --sysroot=$ANDROID_STANDALONE_SYSROOT \
+                        -DXCTEST_BUILD_DIR=$SWIFT_ANDROID_BUILD_PATH/xctest-linux-x86_64 \
+                        -DLIBDISPATCH_SOURCE_DIR=$SWIFT_PATH/swift-corelibs-libdispatch \
+                        -DLIBDISPATCH_BUILD_DIR=$SWIFT_PATH/swift-corelibs-libdispatch &&
+
+                cp -r /usr/include/uuid $ANDROID_STANDALONE_SYSROOT/usr/include 
+                sed -ie "s/-I.\/ -I\/usr\/include\/x86_64-linux-gnu  -I\/usr\/include\/x86_64-linux-gnu    -I\/usr\/include\/libxml2//" build.ninja
```

Comment:
(From above) if this was `sed -i~` instead, you would generate `build.ninja~` as the backup file, which would automatically be excluded.

Reply:
How about if I just do "sed -i" will it edit without backup? I don't need to generate backup.

Reply:
@amraboelela Yes it should do, from the manage:
```
 -i[SUFFIX], --in-place[=SUFFIX]
             edit files in place (makes backup if extension supplied)
```

Reply:
I've seen some versions of using `sed -i` when you don't explicitly specify an extension, but it then takes the following word from the command line and uses that as the extension itself. I would recommend using `sed -i~` so that the backups aren't committed to git (the `*~` pattern is already in the `.gitattributes` file) and it avoids any versions of `sed` where running `sed -i something` doesn't get translated as "use something as the backup extension".

Reply:
This still looks outstanding from GitHub's diffs?